### PR TITLE
allow users to provide feedback

### PR DESCRIPTION
**Pull request by vabarbosa**
# Description

this PR add support for users to provide feedback (via codelens and statusbar).
 in addition, it does some refactoring, clean up of old code.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

manually tested providing 
- positive, negative feedback as well as comments (via codelens)
- feedback comments (via statusbar)

and confirmed data was properly received by the server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/commands/acceptDisclaimer.ts [hunk 1]
```diff
@@ -23,8 +22,6 @@ async function handler(model: ModelInfo): Promise<void> {
   const context = getExtensionContext();
   if (!context) return;
 
-  await requiresToken(context);
```
Comment (cbjuan) [inline src/commands/acceptDisclaimer.ts:26]: why is not required the token anymore in this file?

File: src/commands/selectModel.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import vscode, { ExtensionContext } from "vscode";
 
 import { getExtensionContext } from "../globals/extensionContext";
-import { getModel, getModels } from "../services/codeAssistant";
+import { getServiceApi } from "../services/common";
 import { setDefaultStatus, setLoadingStatus } from "../statusBar/statusBar";
 import { requiresToken } from "../utilities/guards";
```
Comment (cbjuan) [inline src/commands/selectModel.ts:6]: `requiresToken` is not used in the file, the import is not needed

**Review by cbjuan — COMMENTED** [review]
Overall looks good to me. I've made a couple of comments that apply to more files, please, check them out

I've not tested, I'll try to do it before merging

**Comment by cbjuan** [comment]
Could you review this PR, @ajbozarth?

File: src/commands/acceptDisclaimer.ts (hunk 1 above)
Comment (vabarbosa) [reply src/commands/acceptDisclaimer.ts:26]: thinking ahead for the local mode where we dont need a token, so moved it out of the individual files and the check for token now happens when the service is instantiated: https://github.com/Qiskit/qiskit-code-assistant-vscode/pull/102/files#diff-3dabf155febbf2775b0b04a4b659514541a8cba8e2674c4d2b0a8b3bc41f8966R9-R10

File: src/commands/selectModel.ts (hunk 2 above)
Comment (vabarbosa) [reply src/commands/selectModel.ts]: thanks for catching.. [removed](https://github.com/Qiskit/qiskit-code-assistant-vscode/pull/102/commits/c496df446c8e73445568667d07acadabb22d9f26)

File: src/commands/acceptDisclaimer.ts (hunk 1 above)
Comment (cbjuan) [reply src/commands/acceptDisclaimer.ts:26]: got it, thanks!

**Comment by ajbozarth** [comment]
I did a read through of the code and wow that was a lot of changes. IIUC most of the updates are actually a refactor to more easily support the CodeLens addition, specifically it looks like you did something similar to what I did where you made the code completion code extend the VSCode classes and methods rather that just leveraging them in your own classes and functions.

I didn't notice any glaring code issues, but I did notice you have no CI for this repo making sure the code builds and passes a style lint like the jupyterlab repo does. I think adding CI would improve future PR reviews drastically, but I'm unsure how straight-forward adding it is for python libs and VSCode extensions. The JupyterLab extension template includes basic CI out of the box, so I didn't have to write any.

**Comment by vabarbosa** [comment]
> I didn't notice any glaring code issues, but I did notice you have no CI for this repo making sure the code builds and passes a style lint like the jupyterlab repo does. I think adding CI would improve future PR reviews drastically, but I'm unsure how straight-forward adding it is for python libs and VSCode extensions. The JupyterLab extension template includes basic CI out of the box, so I didn't have to write any.

yes, definitely need to add to the TODO

**Review by cbjuan — APPROVED** [review]
thanks!

**Pull request merged**